### PR TITLE
refactor: removed boost dependency

### PR DESCRIPTION
**Pull request by rodolforg**
implement our own Runge-Kutta54 Cash-Karp, that boost::ode
uses in integrate() as claimed in documentation:
https://www.boost.org/doc/libs/1_66_0/libs/numeric/odeint/doc/html/boost_numeric_odeint/getting_started/short_example.html

Algorithm described here
http://www.elegio.it/mc2/rk/doc/p201-cash-karp.pdf

As far as I tested, it gives the same results.

### CODE REVIEW

File: synfig-core/src/synfig/valuenodes/CMakeLists.txt [hunk 1]
```diff
@@ -70,7 +70,7 @@ target_sources(libsynfig
 )
 
 file(GLOB VALUENODES_HEADERS "${CMAKE_CURRENT_LIST_DIR}/*.h")
-target_include_directories(libsynfig PRIVATE ${Boost_INCLUDE_DIRS})
+target_include_directories(libsynfig)
```
Comment (ice0) [inline synfig-core/src/synfig/valuenodes/CMakeLists.txt:73]: Just remove the whole line

**Comment by ice0** [comment]
@morevnaproject 

I created a file with some random values using a "dynamic" converter and tested it with the current PR and `master` branch.
[test-dynamic.zip](https://github.com/synfig/synfig/files/9332720/test-dynamic.zip)

`cmp -l ./test-dynamic.avi ./test-dynamic-boost.avi` shows no difference.

Do you plan to test it yourself or can we merge it?

**Comment by morevnaproject** [comment]
Thank you! I am going to test it too now. ^__^

**Comment by morevnaproject** [comment]
I have finished testing and it gives exactly the same results for me too! Awesome work.

I also checked examples from this page -
https://wiki.synfig.org/Convert#Dynamic

Please rebase and we are ready to merge!

**Comment by rodolforg** [comment]
rebased

File: autobuild/build.sh [hunk 2]
```diff
@@ -214,12 +214,9 @@ cd synfig-core
 pushd "${REPO_DIR}/synfig-core/" >/dev/null
 /bin/bash "${REPO_DIR}/synfig-core/bootstrap.sh"
 popd >/dev/null
-if [ -e /etc/debian_version ] && [ -z "$BOOST_CONFIGURE_OPTIONS" ]; then
+if [ -e /etc/debian_version ]; then
 	# Debian/Ubuntu multiarch
 	MULTIARCH_LIBDIR="/usr/lib/`uname -m`-linux-gnu/"
```
Comment (rodolforg) [inline autobuild/build.sh:219]: Is `MULTIARCH_LIBDIR` still needed after this PR?

File: synfig-core/src/CMakeLists.txt [hunk 3]
```diff
@@ -3,7 +3,7 @@
 ##
 
 # TODO: check if we really need all of them
```
Comment (rodolforg) [inline synfig-core/src/CMakeLists.txt:5]: Is this comment still needed after this PR?

File: autobuild/build.sh (hunk 2 above)
Comment (morevnaproject) [reply autobuild/build.sh:219]: `MULTIARCH_LIBDIR` not needed anymore :+1:

File: synfig-core/src/CMakeLists.txt (hunk 3 above)
Comment (morevnaproject) [reply synfig-core/src/CMakeLists.txt:5]: I believe it is safe to remove. @ice0, are you okay with that? ^__^

Comment (ice0) [reply synfig-core/src/CMakeLists.txt:5]: Sure, this can be removed now.

**Comment by rodolforg** [comment]
Done :)

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**